shared/js/content-scripts/element-hiding.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * The purpose of this content script is to hide third-party tracking ads that have been blocked
+ * and appear as blank spaces on websites. In order to distinguish between tracking ads and legitimate
+ * ads, it is necessary to inject this script into all frames.
+ */
+
+var contentScript = {}
+contentScript.frameType = window === window.top ? 'main' : window.parent === window.top ? 'topLevelFrame' : 'nestedFrame'
+contentScript.possibleTargets = []
+contentScript.foundScripts = []
+contentScript.foundFrames = []
+
+/**
+ * Set up listener in each frame to respond to messages from other frames.
+ * There are three types of messages being passed:
+ * 1. iframes send messages to the main frame requesting their id (frameIdRequest)
+ * 2. main frame sends messages back to iframes with their id (setFrameId)
+ * 3. iframes send messages to main frame when they contain blocked elements (hideFrame)
+ */
+contentScript.frameListener = (e) => {
+    if (e.data.type === 'frameIdRequest') {
+        contentScript.foundFrames = document.getElementsByTagName('iframe')
+        let i = contentScript.foundFrames.length
+        while (i--) {
+            let frame = contentScript.foundFrames[i]
+            if (frame.src && frame.id && !e.data.blockedRequests.includes(frame.src) && !frame.className.includes('ddg-hidden')) {
```

Comment:
Can't inject content scripts into blocked iframes, so no point messaging them. 

Also no point messaging iframes that have already been hidden. My first thought was to add a class to every frame that had already been messaged like this, and not message them again. Unfortunately, some ad scripts dynamically change the src url of iframes, which means this has to be run again.